### DISCUSSION TITLE
Popstate event not triggering

### DISCUSSION THREAD

**Discussion by lightyearsagency** [swup/swup#497] [Help] [answered]
Hi gang.

We've got a client site that's been launched, but I'm wanting to hook a little clean-up function when people navigate away from a certain page.
https://www.letsfcancer.com/cancer-risk-calculator/
(Please ignore the slab of console logs, it's there for debugging)

When a user navigates away and comes back I'm wanting to run a javascript function.
However, the popState event from SWUP isn't working.
I looked through the docs, but couldn't spot any settings I'm missing.

I've got this in my main SWUP function.

```
swup.on('popState', function() {
	console.log("Load from popState"); // Trying to see if this is functioning
	window.scrollTo({top: 0,left: 0,behavior: 'instant'});
	resetCalculator(); // The function I'm trying to run
});
```

However it doesn't work on back\forward.

Any ideas?

**Comment by daun** [comment, marked as answer]
It's working fine for me on the domain you posted. I can see both the `swup:popState` event from the debug plugin as well as your console log message.
